README.md
# Victory Pie Native

```sh
$ npm install
$ npm start
```

Then open `demo/ios/Demo.xcodeproj` and run the app. (Note: `npm start` must run _before_ you run the app).

## Issues 
To make it easier to manage issues across all of Victory, we have disabled issues for this repo. [Please open issues in the main victory-native repo instead](https://github.com/FormidableLabs/victory-native/issues). You can track our progress on issues [here](https://github.com/FormidableLabs/victory-native/projects/1)


## _IMPORTANT_

This project is in a pre-release state. We're hard at work fixing bugs and improving the API. Be prepared for breaking changes!

## Contributor Covenant Code of Conduct

Please review our [Code of Conduct][code] before contributing.

[code]: https://github.com/FormidableLabs/builder-victory-component/blob/master/CONTRIBUTING.md#contributor-covenant-code-of-conduct


CHANGELOG.md
# VictoryPieNative Changelog

## 1.1.1 (2017-06-14)

- Stricter dependency on victory-pie, ~7.1.2; eliminate multiple victory-core's being installed

## 1.1.0 (2016-10-18)

- Update dependencies

## 1.0.0 (2016-08-22)

- **Moves `react-native-svg` to `peerDependencies`**
- Adds tests and updates dependencies

## 0.2.0 (2016-07-29)

- Update dependencies

## 0.1.4 (2016-07-06)

- Update dependencies

## 0.1.3 (2016-07-02)

- Initial working version
- Known issues:
  - Transitions are not working perfectly
  - Events are not supported
  - Poor Android support


test/compiler.js
var fs = require("fs");
var babel = require("babel-core");
var origJs = require.extensions[".js"];

// Mock react-native and react-native-svg
require("react-native-mock/mock");
require("react-native-svg-mock/mock");

// Compile a path with babel
var compile = function (fileName) {
  var src = fs.readFileSync(fileName, "utf8");
  var output = babel.transform(src, {
    filename: fileName
  }).code;
  return output;
}

// We need to compile a few node_modules
var shouldCompile = function (fileName) {
  var should = [
    "node_modules/victory-core-native",
    "node_modules/victory-core",
    "node_modules/victory-pie"
  ].some(function (mod) {
    return fileName.indexOf(mod) >= 0;
  });
  return should;
};

// Babel compile things not in node_modules
require.extensions[".js"] = function (module, fileName) {
  if (shouldCompile(fileName) || fileName.indexOf("node_modules/") === -1) {
    return module._compile(compile(fileName), fileName);
  }
  return origJs(module, fileName);
};


test/spec/test.js
import React from "react-native";
import { render } from "enzyme";
import { expect } from "chai";

import VictoryPie from "../../lib/components/victory-pie";

const components = [
  { component: VictoryPie, name: "VictoryPie" }
];

describe("Default render", () => {
  components.forEach((c) => {
    it(`should work for ${c.name}`, () => {
      const wrapper = render(React.createElement(c.component));
      expect(wrapper).to.have.length(1);
    });
  });
});


demo/index.ios.js
/* global setInterval */
/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import random from "lodash.random";
import range from "lodash.range";
import React, { Component } from "react";
import {
  AppRegistry,
  ScrollView,
  StyleSheet,
} from "react-native";
import { VictoryPie } from "../lib";

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: "#e1d7cd"
  }
});

class Demo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: this.generateRandomData()
    };
  }
  componentWillMount() {
    setInterval(() => {
      this.setState({data: this.generateRandomData()});
    }, 2000);
  }
  generateRandomData() {
    return range(1, 7).map(() => ({x: " ", y: random(1, 10)}));
  }
  render() {
    return (
      <ScrollView contentContainerStyle={styles.container}>
        <VictoryPie
          innerRadius={75}
          data={this.state.data}
          animate={{
            duration: 800,
          }}
        />

        <VictoryPie
          style={{
            labels: {
              fontSize: 15,
              padding: 75
            }
          }}
        />
        <VictoryPie
          style={{
            data: {
              stroke: "none",
              opacity: 0.3
            }
          }}
        />
        <VictoryPie innerRadius={90} />
        <VictoryPie
          endAngle={90}
          startAngle={-90}
        />
        <VictoryPie
          endAngle={90}
          innerRadius={90}
          padAngle={5}
          startAngle={-90}
        />
        <VictoryPie
          style={{
            labels: {
              fill: "white",
              stroke: "none",
              fontSize: 10,
              fontWeight: "bold"
            }
          }}
          data={[
            {x: "<5", y: 6279},
            {x: "5-13", y: 9182},
            {x: "14-17", y: 5511},
            {x: "18-24", y: 7164},
            {x: "25-44", y: 6716},
            {x: "45-64", y: 4263},
            {x: "≥65", y: 7502}
          ]}
          innerRadius={70}
          colorScale={[
            "#D85F49",
            "#F66D3B",
            "#D92E1D",
            "#D73C4C",
            "#FFAF59",
            "#E28300",
            "#F6A57F"
          ]}
        />
        <VictoryPie
          style={{
            data: {
              stroke: (data) => data.y > 75 ? "black" : "none",
              opacity: (data) => data.y > 75 ? 1 : 0.4
            }
          }}
          data={[
            {x: "Cat", y: 62},
            {x: "Dog", y: 91},
            {x: "Fish", y: 55},
            {x: "Bird", y: 55}
          ]}
        />
      </ScrollView>
    );
  }
}

AppRegistry.registerComponent("Demo", () => Demo);


demo/index.android.js
/* global setInterval */
/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 */

import random from "lodash.random";
import range from "lodash.range";
import React, { Component } from "react";
import {
  AppRegistry,
  ScrollView,
  StyleSheet,
} from "react-native";
import { VictoryPie } from "../lib";

VictoryPie.defaultProps = {
  ...VictoryPie.defaultProps,
  height: 300,
  width: 300
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: "#e1d7cd"
  }
});

class Demo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      data: this.generateRandomData()
    };
  }
  componentWillMount() {
    setInterval(() => {
      this.setState({data: this.generateRandomData()});
    }, 2000);
  }
  generateRandomData() {
    return range(1, 7).map(() => ({x: null, y: random(1, 10)}));
  }
  render() {
    return (
      <ScrollView contentContainerStyle={styles.container}>
        <VictoryPie
          innerRadius={75}
          data={this.state.data}
          animate={{
            duration: 800,
            data: this.state.data
          }}
        />
        <VictoryPie
          data={[
            {animal: "Cat", pet: 45, wild: 17},
            {animal: "Dog", pet: 85, wild: 6},
            {animal: "Fish", pet: 55, wild: 0},
            {animal: "Bird", pet: 15, wild: 40}
          ]}
          x="animal"
          y={(data) => data.pet + data.wild}
        />
        <VictoryPie
          style={{
            labels: {
              fontSize: 15,
              padding: 75
            }
          }}
        />
        <VictoryPie
          style={{
            data: {
              stroke: "none",
              opacity: 0.3
            }
          }}
        />
        <VictoryPie innerRadius={90} />
        <VictoryPie
          endAngle={90}
          startAngle={-90}
        />
        <VictoryPie
          endAngle={90}
          innerRadius={90}
          padAngle={5}
          startAngle={-90}
        />
        <VictoryPie
          style={{
            labels: {
              fill: "white",
              fontSize: 10,
              fontWeight: "bold"
            }
          }}
          data={[
            {x: "<5", y: 6279},
            {x: "5-13", y: 9182},
            {x: "14-17", y: 5511},
            {x: "18-24", y: 7164},
            {x: "25-44", y: 6716},
            {x: "45-64", y: 4263},
            {x: "≥65", y: 7502}
          ]}
          innerRadius={70}
          colorScale={[
            "#D85F49",
            "#F66D3B",
            "#D92E1D",
            "#D73C4C",
            "#FFAF59",
            "#E28300",
            "#F6A57F"
          ]}
        />
        <VictoryPie
          style={{
            data: {
              stroke: (data) => data.y > 75 ? "black" : "transparent",
              opacity: (data) => data.y > 75 ? 1 : 0.4
            }
          }}
          data={[
            {x: "Cat", y: 62},
            {x: "Dog", y: 91},
            {x: "Fish", y: 55},
            {x: "Bird", y: 55}
          ]}
        />
      </ScrollView>
    );
  }
}

AppRegistry.registerComponent("Demo", () => Demo);
